src/reconcile/rec_visibility.c

```diff
@@ -252,6 +257,112 @@ __timestamp_out_of_order_fix(WT_SESSION_IMPL *session, WT_TIME_WINDOW *select_tw
     return (false);
 }
 
+/*
+ * __rec_validate_upd_chain --
+ *     Check the update chain for conditions that would prevent its insertion into the history
+ *     store. Return EBUSY if the update chain cannot be inserted into the history store at this
+ *     time.
+ */
+static int
+__rec_validate_upd_chain(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_UPDATE *select_upd,
+  WT_TIME_WINDOW *select_tw, WT_CELL_UNPACK_KV *vpack)
+{
+    WT_DECL_RET;
+    WT_UPDATE *prev_upd, *upd;
+
+    /*
+     * There is no selected update to go to disk as such we don't need to check the updates
+     * following it.
+     */
+    if (select_upd == NULL)
+        return (0);
+
+    /* There is no history store so it is safe to reconcile out of order updates. */
+    if (!F_ISSET(S2C(session), WT_CONN_HS_OPEN))
+        return (0);
+
+    /* History store out of order update is allowed. */
+    if (WT_IS_HS(session->dhandle))
+        return (0);
+
+    /*
+     * If we're not in eviction any history store insertion is fine, if we are in eviction and a
+     * checkpoint isn't running then history store insertions are also fine.
```

Comment:
I've been confused by this flag a number of times - it means a checkpoint wasn't running at the start of the reconciliation, not that there isn't one running now. Could you update the comment to be more accurate please?

Reply:
Updated the comment.